src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapAction_JsonBodyOrService_HandlesBothJsonAndService.generated.txt

```diff
@@ -94,8 +94,21 @@ namespace Microsoft.AspNetCore.Http.Generated
             [(@"TestMapActions.cs", 25)] = (
                 (methodInfo, options) =>
                 {
-                    Debug.Assert(options?.EndpointBuilder != null, "EndpointBuilder not found.");
+                    Debug.Assert(options != null, "RequestDelegateFactoryOptions not found.");
+                    Debug.Assert(options.EndpointBuilder != null, "EndpointBuilder not found.");
                     options.EndpointBuilder.Metadata.Add(new SourceKey(@"TestMapActions.cs", 25));
+                    var serviceProvider = options.ServiceProvider ?? options.EndpointBuilder.ApplicationServices;
+                    var serviceProviderIsService = serviceProvider.GetRequiredService<IServiceProviderIsService>();
+                    var jsonBodyOrServiceTypeTuples = new (bool, Type)[] {
+                        (false, typeof(global::Microsoft.AspNetCore.Http.Generators.Tests.Todo)),
+                        (false, typeof(global::Microsoft.AspNetCore.Http.Generators.Tests.TestService)),
+                    };
+                    var inferredBodyParameters = jsonBodyOrServiceTypeTuples.Where(p => !serviceProviderIsService.IsService(p.Item2));
```

Comment:
Why not a foreach where we emit for the first parameter that we find (if any) that is not derived from body.

```csharp
foreach (var (isOptional, type) in jsonBodyOrServiceTypeTuples)
{
  if (!serviceProviderIsService.IsService(type))
  {
    options.EndpointBuilder.Metadata.Add(...);
    break;
  }
}
```

Reply:
Done!